test/webhook-apicoverage/tools/tools.go

```diff
@@ -186,3 +191,56 @@ func GetAndWriteTotalCoverage(webhookIP string, outputFile string) error {
 
 	return nil
 }
+
+// GetResourcePercentages calls resource percentage coverage API to retrieve
+// percentage values.
+func GetResourcePercentages(webhookIP string) (
+	*coveragecalculator.CoveragePercentages, error) {
+	client := &http.Client{Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	},
+	}
+
+	resp, err := client.Get(fmt.Sprintf(WebhookResourcePercentageCoverageEndPoint,
+		webhookIP))
+	if err != nil {
+		return nil, fmt.Errorf(
+			"encountered error making resource percentage coverage request:" +
+				" %v", err)
+	} else if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("invalid HTTP Status received for resource" +
+			" percentage coverage request. Status: %d", resp.StatusCode)
+	}
+
+	var body []byte
+	if body, err = ioutil.ReadAll(resp.Body); err != nil {
+		return nil, fmt.Errorf("error reading resource percentage" +
+			" coverage response: %v", err)
+	}
+
+	var coveragePercentages coveragecalculator.CoveragePercentages
```

Comment:
```suggestion
	coveragePercentages := &coveragecalculator.CoveragePercentages{}
```
And remove pointer logic below.

Reply:
Fixed.